### PR TITLE
New mode view Kanban for list expeditions

### PR DESCRIPTION
**Pull request by lamrani002**
# NEW| mode view "kanban" for Expedition list
NB: add function "getKanbanView()" in abstract class "CommonobjectLine" with needed informations

### CODE REVIEW

File: htdocs/core/lib/project.lib.php [hunk 1]
```diff
@@ -1064,6 +1064,239 @@ function projectLinesa(&$inc, $parent, &$lines, &$level, $var, $showproject, &$t
...
+ * Output a task line into a pertime intput mode
+ *
+ * @param	string	   	$inc					Line number (start to 0, then increased by recursive call)
+ * @param   string		$parent					Id of parent task to show (0 to show all)
+ * @param	User|null	$fuser					Restrict list to user if defined
+ * @param   Task[]		$lines					Array of lines
+ * @param   int			$level					Level (start to 0, then increased/decrease by recursive call)
+ * @param   string		$projectsrole			Array of roles user has on project
+ * @param   string		$tasksrole				Array of roles user has on task
+ * @param	string		$mine					Show only task lines I am assigned to                                              <<<<<<<<< Temporary merge branch 1
```
Comment (eldy) [inline htdocs/core/lib/project.lib.php:1077]: Something wrong here. We found a line
<<<<<<<<< Temporary merge branch 1

File: htdocs/core/lib/project.lib.php [hunk 2]
```diff
@@ -1064,6 +1064,239 @@ function projectLinesa(&$inc, $parent, &$lines, &$level, $var, $showproject, &$t
 }
 
 
+/**
```
Comment (eldy) [inline htdocs/core/lib/project.lib.php:1067]: Something wrong here, we should not have this file modified in this PR

File: htdocs/core/class/commonobjectline.class.php [hunk 3]
```diff
@@ -166,4 +166,37 @@ public function fetchAll($sortorder = '', $sortfield = '', $limit = 0, $offset =
...
+
+		/**
+	 *	Return clicable link of object (with eventually picto)
+	 *
+	 *	@param      string	    $option                 Where point the link (0=> main card, 1,2 => shipment, 'nolink'=>No link)
+	 *  @return		string		HTML Code for Kanban thumb.
+	 */
+	public function getKanbanView($option = '')
+	{
+		global $db, $langs;
```
Comment (eldy) [inline htdocs/core/class/commonobjectline.class.php:178]: Inside a class, do not use global $db
You already have $this->db available

File: htdocs/core/lib/project.lib.php (hunk 2 above)
Comment (lamrani002) [reply htdocs/core/lib/project.lib.php]: I pull develop into my branch KanbanExpedition and resolve the conflict in project.lib.php.
It should work properly

**Pull request merged**